sdk/iOS/test/MSManagedObjectObserverTests.m

```diff
@@ -0,0 +1,176 @@
+//
+//  MSManagedObjectObserverTests.m
+//  WindowsAzureMobileServices
+//
+//  Created by Damien Pontifex on 13/03/2015.
+//  Copyright (c) 2015 Windows Azure. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "MSCoreDataStore.h"
+#import "MSCoreDataStore+TestHelper.h"
+#import "MSJSONSerializer.h"
+#import "TodoItem.h"
+#import "MSManagedObjectObserver.h"
+#import "MSTableOperation.h"
+
+@interface MSManagedObjectObserverTests : XCTestCase
+@property (nonatomic, strong) MSClient *client;
+@property (nonatomic, strong) MSCoreDataStore *store;
+@property (nonatomic, strong) NSManagedObjectContext *context;
+@property (nonatomic, strong) MSManagedObjectObserver *observer;
+
+@property (nonatomic, strong) TodoItem *item;
+@end
+
+@implementation MSManagedObjectObserverTests
```

Comment:
Ideally we should have a test case that results in an error to make sure it surfaces correctly.  Perhaps manually put an insert operation for the item to be inserted already in the operation table, that should cause it to error out.